worlds/yoshisisland/level_logic.py

```diff
@@ -11,46 +15,44 @@ class YoshiLogic:
     consumable_logic: bool
     luigi_pieces: int
 
-
-    def __init__(self, world: Optional["YoshisIslandWorld"]) -> None:
+    def __init__(self, world: "YoshisIslandWorld") -> None:
         self.player = world.player
         self.boss_order = world.boss_order
-        self.luigi_pieces = world.luigi_pieces
+        self.luigi_pieces = world.options.luigi_pieces_required.value
 
-        if world:
-            if world.options.stage_logic == 0:
-                self.game_logic = "Easy"
-            elif world.options.stage_logic == 1:
-                self.game_logic = "Normal"
-            else:
-                self.game_logic = "Hard"
+        if world.options.stage_logic == StageLogic.option_strict:
+            self.game_logic = "Easy"
+        elif world.options.stage_logic == StageLogic.option_loose:
+            self.game_logic = "Normal"
+        else:
+            self.game_logic = "Hard"
 
-            self.midring_start = world.options.shuffle_midrings == 0
-            self.consumable_logic = world.options.item_logic != 0
+        self.midring_start = not world.options.shuffle_midrings
+        self.consumable_logic = not world.options.item_logic
 
-            if world.options.hidden_object_visibility >= 2:
-                self.clouds_always_visible = True
-            else:
-                self.clouds_always_visible = False
+        if world.options.hidden_object_visibility >= ObjectVis.option_clouds_only:
+            self.clouds_always_visible = True
+        else:
+            self.clouds_always_visible = False
 
-            self.bowser_door = world.options.bowser_door_mode
-            if self.bowser_door not in {0, 1, 2, 5}:
-                self.bowser_door = 3
+        self.bowser_door = world.options.bowser_door_mode.value

```

Comment:
This code seems wrong, before and after this PR. Either 3 and 4 become 3 or only 4 becomes 3, both of which seem very strange.

Reply:
If 4 becomes 3 and 3 is itself, then that's the same as 3 and 4 becoming 3, no?

Reply:
So at least it's consistent with the old code I mean.

Reply:
Oh yeah, you're right. It just looks weird I guess

Reply:
Door 3 and 4 both have no logic; it was decided that instead of having 2 logic functions that both returned True, it was better to have 1 function and have option 4 redirect to 3 instead.